### PR TITLE
Add Linting to the Demo Project

### PR DESCRIPTION
**Pull request by abdulmth**
Add prettier and the `format` command and run it. 
Fixed some missing html tags.

Edit: 
* Added a full linting for JS, TS and HTML files. 
* Added CI workflow that runs the lint command.

### CODE REVIEW

File: demo-project/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
```
Comment (ntn-x2) [inline demo-project/.prettierrc:2]: Personal preference, do we need this?

File: demo-project/package.json [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
-        "@types/express-jwt": "^7.4.2",
-        "parcel": "^2.9.3"
-    }
+  "name": "demo-project",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "scripts": {
+    "build": "parcel build main.ts",
+    "format": "npx prettier --write ."
```
Comment (ntn-x2) [inline demo-project/package.json:8]: We also need another script to check for lint errors in the CI, as we do here: https://github.com/KILTprotocol/sdk-js/blob/23c5b045dc7d3d587aeca5f6650736bb1218130b/.github/workflows/tests.yml#L73.

**Review by ntn-x2 — CHANGES_REQUESTED** [review]
Can you please add a CI step that checks if the lint rules are followed?

File: demo-project/.prettierrc (hunk 1 above)
Comment (abdulmth) [reply demo-project/.prettierrc:2]: It's the same option in the frontend, I wanted to keep it consistent.

Comment (ntn-x2) [reply demo-project/.prettierrc:2]: I would remove it from the frontend as well, but don't wanna hang too much on these comments, so I'll resolve it. Feel free to change if you like.

File: demo-project/package.json [hunk 3]
```diff
@@ -3,8 +3,18 @@
...
+    },
     "scripts": {
-        "build": "parcel build main.ts"
+        "build": "parcel build main.ts",
+        "format": "npx prettier --write .",
+        "lint": "yarn lint-backend && yarn lint-frontend",
+        "lint-backend": "eslint --max-warnings 0 --config ./.eslintrc.json  \"./**/*.ts*\"",
+        "lint-frontend": "eslint --max-warnings 0 --config ./.eslintrc.json  \"./demo-frontend/**/*\"",
+        "fix": "yarn fix-backend && yarn fix-frontend",
+        "fix-backend": "eslint --max-warnings 0 --fix --config ./.eslintrc.json  \"./**/*.ts*\"",
```
Comment (ntn-x2) [inline demo-project/package.json:16]: This should be `yarn lint-backend --fix`, right?

File: .github/workflows/demo-project.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        working-directory: demo-project
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+
+      - name: Configure Node and package manager
+        uses: actions/setup-node@v3.6.0
+        with:
+          node-version-file: "demo-project/package.json"
```
Comment (ntn-x2) [inline .github/workflows/demo-project.yml:17]: What does it mean if the node requirement is to be >= than a value? Which version does it pick? We've used `.nvmrc` files everywhere else, maybe we should stick to those here as well? Let's not deviate too much between repos.

File: demo-project/package.json (hunk 3 above)
Comment (abdulmth) [reply demo-project/package.json]: [d8f68f6](https://github.com/KILTprotocol/opendid/pull/55/commits/d8f68f65beb66003ce6cb05aceb52aa50db830fa)

**Review by ntn-x2 — APPROVED** [review]
Should not rely only on my review for web and TS stuff 😄

File: demo-project/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            ],
+            "parser": "@html-eslint/parser",
+            "extends": [
+                "plugin:@html-eslint/recommended"
+            ]
+        },
+        {
+            "files": [
+                "*.js"
+            ],
```
Comment (Dudleyneedham) [inline demo-project/.eslintrc.json:40]: Shame we have a `js` file. Would it be too much to convert this to a `ts` so we can remove this `js` config part?

File: demo-project/.prettierrc (hunk 1 above)
Comment (Dudleyneedham) [inline demo-project/.prettierrc:2]: Can we set to 'false' even though it converts it?

File: demo-project/package.json [hunk 6]
```diff
@@ -3,8 +3,17 @@
     "version": "1.0.0",
     "main": "index.js",
     "license": "MIT",
+    "engines": {
+        "node": ">= v18.17.1"
```
Comment (Dudleyneedham) [inline demo-project/package.json:7]: Can we convert this to version 20 in another PR?

**Review by Dudleyneedham — APPROVED** [review]
LGTM some minor comments

File: demo-project/.eslintrc.json (hunk 5 above)
Comment (abdulmth) [reply demo-project/.eslintrc.json:41]: It's injected in the frontend directly, converting it to TS would need a trsnapile step.

File: demo-project/.prettierrc (hunk 1 above)
Comment (abdulmth) [reply demo-project/.prettierrc]: [9111a16](https://github.com/KILTprotocol/opendid/pull/55/commits/9111a169e73701e8e776465bd0b264cea312aba5)

File: demo-project/.eslintrc.json (hunk 5 above)
Comment (Dudleyneedham) [reply demo-project/.eslintrc.json:41]: Boo that is annoying... Ha

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**